### DISCUSSION TITLE
Snowflake connection timeout

### DISCUSSION THREAD

**Discussion by mielune** [Budibase/budibase#7403] [Help] [answered]
Hi,

I'm building a new app trying to use the new Snowflake connector, but it failed with 10sec timeout when running the 1st query.
We are using a self-hosted instance of BB v1.2.20.

We have a Snowflake account available at an url through a private link like this: kt0000.west-europe.privatelink.snowflakecomputing.com
I tried with `kt000000` or `kt0000.west-europe.privatelink` in the account field, but same result.
I also tried with the "org_name-account_name" and "org_name-account_name.privatelink", same result.

No details in the app service logs (see attached screenshots).

No traces of the calls in the Snowflake logs.

Have you any idea of the trouble ? It seems that the api serveur url is not 

![2022-08-22 15_25_28-](https://user-images.githubusercontent.com/8350742/185933472-00da17a0-3af7-484f-8654-acc8f441c887.png)

![image](https://user-images.githubusercontent.com/8350742/185934122-4fda5d8e-cf3e-4259-a2c5-16d59db88f92.png)

**Comment by melohagan** [comment]
Hi @mielune 

Could you confirm that you are able to successfully connect and run a query through the SnowSQL CLI?

**Comment by mielune** [comment]
Hi,

Yes I can.
Next week, I will look with a snowflake admin at work.
I will send news here.

**Comment by mielune** [comment, marked as answer]
Hi @melohagan 

We finally found the root cause of this error:

The snowflake accounts named "xxxxx.west-europe.privatelink" are dns entries on Azure (we are on it for Snowflake).

They are available both externally in Internet and throw the private link between our internal networks. The public IP linked to it has no open ports (it is for private network usage) and throw the private ports are open.
The trouble was our default DNS service used by our docker cluster that was not doing the DNS resolution throw the internal link, but external one.

Thank you for your support and your awesome product ;)

**Reply by melohagan** [reply]
Really glad to hear you resolved the issue, thanks very much for sharing your solution!
